shotover-proxy/src/protocols/cassandra_protocol2.rs

```diff
@@ -646,132 +646,232 @@ impl Encoder<Messages> for CassandraCodec2 {
+        TableFactor, TableWithJoins, Value as SQLValue,
+    };
     use std::collections::HashMap;
     use tokio_util::codec::{Decoder, Encoder};
 
-    const STARTUP_BYTES: [u8; 31] =
-        hex!("0400000001000000160001000b43514c5f56455253494f4e0005332e302e30");
-
-    const READY_BYTES: [u8; 9] = hex!("840000000200000000");
-
-    const REGISTER_BYTES: [u8; 58] = hex!(
-        "040000010b000000310003000f544f504f4c4f47595f4348414e4745
-    000d5354415455535f4348414e4745000d534348454d415f4348414e4745"
-    );
-
-    const QUERY_BYTES: [u8; 60] = hex!(
-        "0400000307000000330000002c53454c454354202a2046524f4d20737973
-    74656d2e6c6f63616c205748455245206b65793d276c6f63616c27000100"
-    );
-
-    const RESULT_BYTES: [u8; 162] = hex!(
-        "840000020800000099000000020000000100000009000673797374656
-    d000570656572730004706565720010000b646174615f63656e746572000d0007686f73745f6964000c000c70726566
-    65727265645f6970001000047261636b000d000f72656c656173655f76657273696f6e000d000b7270635f616464726
-    573730010000e736368656d615f76657273696f6e000c0006746f6b656e730022000d00000000"
-    );
-
-    fn test_frame(codec: &mut CassandraCodec2, raw_frame: &[u8]) {
-        let message = codec
+    fn test_frame_codec_roundtrip(
+        codec: &mut CassandraCodec2,
+        raw_frame: &[u8],
+        expected_messages: Vec<Message>,
+    ) {
+        // test decode
+        let decoded_messages = codec
             .decode(&mut BytesMut::from(raw_frame))
             .unwrap()
             .unwrap();
+        assert_eq!(decoded_messages, expected_messages);
 
+        // test encode round trip
         let mut dest = BytesMut::new();
-        codec.encode(message, &mut dest).unwrap();
+        codec.encode(decoded_messages, &mut dest).unwrap();
         assert_eq!(raw_frame, &dest);
     }
 
-    #[test]
-    fn test_startup_codec() {
+    fn new_codec() -> CassandraCodec2 {
         let mut pk_map = HashMap::new();
         pk_map.insert("test.simple".to_string(), vec!["pk".to_string()]);
         pk_map.insert(
             "test.clustering".to_string(),
             vec!["pk".to_string(), "clustering".to_string()],
         );
-        let mut codec = CassandraCodec2::new(pk_map, false);
-        test_frame(&mut codec, &STARTUP_BYTES);
+        CassandraCodec2::new(pk_map, false)
     }
 
     #[test]
-    fn test_ready_codec() {
-        let mut pk_map = HashMap::new();
-        pk_map.insert("test.simple".to_string(), vec!["pk".to_string()]);
-        pk_map.insert(
-            "test.clustering".to_string(),
-            vec!["pk".to_string(), "clustering".to_string()],
-        );
-        let mut codec = CassandraCodec2::new(pk_map, false);
-        test_frame(&mut codec, &READY_BYTES);
+    fn test_codec_startup() {
+        let mut codec = new_codec();
+        let bytes = hex!("0400000001000000160001000b43514c5f56455253494f4e0005332e302e30");
```

Comment:
Wouldn't converting a string to hex be clearer? Instead of having to convert this hex to find out what it's sending.

Reply:
![image](https://user-images.githubusercontent.com/5120858/140465343-9582ceda-fc76-4a09-97ee-7d3cd1f613fb.png)
They are mostly binary data.
We could possibly split them up into documented subsections so we understand what the data is, but that would be out of scope of this PR.